### ISSUE TITLE
Convolution example does not work

### ISSUE THREAD

**Issue by kzhai** [Lasagne/Lasagne#525]
I am trying out CNN model on mnist data as provided by the example.

```
$ python mnist.py cnn
Loading data...
Building model and compiling functions...
Starting training...
Epoch 1 of 500 took 494.369s
  training loss:		2.471961
  validation loss:		2.303229
  validation accuracy:		10.64 %
Epoch 2 of 500 took 493.647s
  training loss:		2.366910
  validation loss:		2.303151
  validation accuracy:		10.64 %
Epoch 3 of 500 took 495.702s
  training loss:		2.332428
  validation loss:		2.301815
  validation accuracy:		10.64 %
Epoch 4 of 500 took 494.423s
  training loss:		2.319300
  validation loss:		2.301806
  validation accuracy:		10.64 %
Epoch 5 of 500 took 498.449s
  training loss:		2.313726
  validation loss:		2.301416
  validation accuracy:		10.64 %
Epoch 6 of 500 took 494.424s
  training loss:		2.310590
  validation loss:		2.301775
  validation accuracy:		10.64 %
...
```

It seems to me the validation loss and accuracy are never changed. I tried to modified learning rates as well as network configuration (size, nonlinearity, etc), and it doesn't seem to work at all. I am wondering what is the problem? Do I still need to wait for more epochs (I wait for 20 epochs and the number does not change at all)?
I also tried the CNN example for Theano, and the validation accuracy reaches around 80-90% within first 3 epochs.

I am using the most updated version of lasagne and theano.

```
Python 2.7.9 (default, Feb 10 2015, 03:28:08) 
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.56)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import theano
>>> print theano.__version__
0.7.0.dev-15c90dd36c95ea2989a2c0085399147d15a6a477
>>> import lasagne
>>> print lasagne.__version__
0.2.dev1
```

**Comment by petrbel** [comment]
@kzhai That's really weird. I download the example about 2 day ago and it converged actually pretty quickly. I use Python 3.4 which shouldn't make any difference, though.

**Comment by kzhai** [comment]
@petrbel May I ask what is the version for theano and lasagne?

**Comment by petrbel** [comment]
@kzhai sure!

```
$ pip3 show numpy scipy theano lasagne
---
Name: numpy
Version: 1.10.1
Location: /usr/local/lib/python3.4/dist-packages
Requires: 
---
Name: scipy
Version: 0.16.1
Location: /usr/local/lib/python3.4/dist-packages
Requires: 
---
Name: Theano
Version: 0.7.0
Location: /usr/local/lib/python3.4/dist-packages
Requires: numpy, scipy, six
---
Name: Lasagne
Version: 0.1
Location: /usr/local/lib/python3.4/dist-packages
Requires: numpy
```

**Comment by kzhai** [comment]
@petrbel Thanks very much! I uninstalled Lasagne 0.2, and reinstalled Lasagne 0.1, and it suddenly works now.
```
python mnist.py cnn
Loading data...
Building model and compiling functions...
Starting training...
Epoch 1 of 500 took 507.908s
  training loss:		1.525951
  validation loss:		0.320161
  validation accuracy:		91.62 %
```

Don't know what was exactly going on with the 0.2 version.

**Comment by petrbel** [comment]
@kzhai Glad to help!

Btw is there Lasagne v0.2 already? [I see](https://github.com/Lasagne/Lasagne/releases) only one release so far.

**Comment by benanne** [comment]
There is no 0.2 release yet, we refer to the current development version as 0.2.dev1, I believe. This is very odd, the example works fine for me with the latest development version. I just tried running it again. Perhaps there was something wrong with your installation.

**Comment by kzhai** [comment]
@petrbel @benanne Right, 0.2.dev1 is what I referred.
I don't know what exactly happened with my installation.
